openpype/hosts/blender/blender_addon/addons/update_paths_to_last_version.py

```diff
@@ -0,0 +1,213 @@
+import os
+import logging
+import re
+
+from openpype.pipeline.anatomy import Anatomy
+from openpype.lib import StringTemplate
+from openpype.pipeline.context_tools import get_template_data_from_session
+
+import bpy
+
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger(__name__)
+
+
+bl_info = {
+    "name": "Update paths to last version",
+    "description": "Update all concerned paths in scene to automatically target the last version existing",
+    "author": "Quad",
+    "version": (1, 0),
+    "blender": (2, 80, 0),
+    "category": "Render",
+    "location": "View 3D > UI",
+}
+
+
+class VIEW3D_PT_UPDATE_PATHS(bpy.types.Panel):
+    bl_label = "Select object types to update"
+    bl_space_type = "VIEW_3D"
+    bl_region_type = "UI"
+    bl_category = "Quad"
+
+    update_alembics = bpy.props.BoolProperty(default=True)
+    update_shader_files = bpy.props.BoolProperty(default=True)
+
+    def draw(self, context):
+        layout = self.layout
+        col = layout.column(align=True)
+        col.operator("paths.update_animation_paths", text="Update alembics to animation", icon="MESH_CUBE").mode = "animation"
+        col.operator("paths.update_animation_paths", text="Update alembics versions", icon="MESH_CUBE").mode = "update_cache_versions"
+
+
+class OBJECT_OT_UPDATE_PATHS(bpy.types.Operator):
+    bl_idname = "paths.update_animation_paths"
+    bl_label = "Update tasks in scene objects paths"
+
+    mode: bpy.props.StringProperty()
+
+    def execute(self, context):
+        mesh_objects = [obj for obj in bpy.data.objects if obj.type == 'MESH']
+
+        if not mesh_objects:
+            log.error("Scene does not contain Mesh objects.")
+            return {'CANCELLED'}
+
+        # Filter modifiers and get cache files directly
+        mesh_sequence_caches = self.get_mesh_sequence_cache_modifiers(mesh_objects)
+        modifiers_cache_files = [modifier.cache_file for modifier in mesh_sequence_caches]
+
+        # Create library overrides in one go
+        self.create_library_override_for(collections=bpy.context.scene.collection,
+                                         objects=mesh_objects,
+                                         modifiers_cache_files=modifiers_cache_files)
+
+        if self.mode == 'update_cache_versions':
+            self.update_versions(bpy.data.cache_files)
+        else:
+            # Update the sequence caches before creating overrides
+            self.update_mesh_sequence_caches(mesh_objects)
+
+            # Set Scale
+            for mesh_object in mesh_objects:
+                mesh_object.scale = (0.01, 0.01, 0.01)
```

Comment:
@hfarre do you know why the scale is set to `0.01` ?

Reply:
Because the scale between maya and blender are not the same, and Guilhem just hardcoded the ratio ^^'
This will be part of the 3D pipe to make a scale converter between softs

Reply:
@ccaillot can you add that as a comment so we know that this scale it to properly import maya elements to Blender

Reply:
done